packages/toggle/Toggle.tsx

```diff
@@ -1,69 +1,79 @@
 import styles from './Toggle.scss';
-import { h, Component, prop, props, emit } from 'skatejs';
+import { h, Component, props, emit } from 'skatejs';
 import {
   ColorType,
   cssClassForColorType,
   css,
-  GenericTypes,
-  GenericEvents
+  GenericEvents,
+  prop,
+  shadyCssStyles
 } from '@blaze-elements/common';
 
-type ToggleProps = Props & EventProps;
-type EventProps = {
+export type ToggleProps = Props & EventProps;
+
+export type Attrs = {
+  disabled?: boolean,
+  checked?: boolean,
+  color?: ColorType,
+};
+
+export type EventProps = {
   onChange?: GenericEvents.CustomChangeHandler<string>,
 };
-type Events = {
+
+export type Events = {
   change?: GenericEvents.CustomChangeHandler<string>,
 };
-type Props = {
+
+export type Props = {
   disabled?: boolean,
   checked?: boolean,
   color?: ColorType,
 };
 
-declare global {
-  namespace JSX {
-    interface IntrinsicElements {
-      'bl-toggle': GenericTypes.IntrinsicCustomElement<ToggleProps> & GenericTypes.IntrinsicBoreElement<Props, Events>
-    }
-  }
-}
+@shadyCssStyles()
+export default class Toggle extends Component<ToggleProps> {
 
-export class Toggle extends Component<ToggleProps> {
-  static get is() { return 'bl-toggle'; }
-  static get props() {
-    return {
-      disabled: prop.boolean( {
-        attribute: true
-      } ),
-      checked: prop.boolean( {
-        attribute: true
-      } ),
-      color: prop.string( {
-        attribute: true
-      } )
-    };
-  }
   static get events() {
     return {
       CHANGE: 'change'
     };
   }
-  disabled = false;
-  checked = false;
-  color: ColorType;
+
+  @prop( {
+    type: Boolean,
+    attribute: {
+      source: true
+    }
+  } ) disabled = false;
```

Comment:
problem is that then it doesn't seems to be working, after removing @prop decorator value is not propagated from attribute to property.